internal/pkg/deploy/cloudformation/stack/env.go

```diff
@@ -189,21 +189,42 @@ func (e *EnvStackConfig) telemetryConfig() *template.Telemetry {
 	}
 }
 
-func (e *EnvStackConfig) importCertARNs() []string {
+func (e *EnvStackConfig) importPublicCertARNs() []string {
 	// If a manifest is present, it is the only place we look at.
 	if e.in.Mft != nil {
 		return e.in.Mft.HTTPConfig.Public.Certificates
 	}
 	// Fallthrough to SSM config.
+	if e.in.ImportCertARNs == nil {
+		return nil
+	}
 	return e.in.ImportCertARNs
 }
 
+func (e *EnvStackConfig) importPrivateCertARNs() []string {
+	// If a manifest is present, it is the only place we look at.
+	if e.in.Mft != nil {
+		return e.in.Mft.HTTPConfig.Private.Certificates
+	}
+	// Fallthrough to SSM config.
+	if e.in.ImportCertARNs == nil {
+		return nil
+	}
+	if e.in.ImportVPCConfig.PublicSubnetIDs == nil {
+		return e.in.ImportCertARNs
+	}
+	return nil
```

Comment:
nit:
```suggestion
	// Fallthrough to SSM config.
	if len(e.in.ImportVPCConfig.PublicSubnetIDs) == 0 {
		return e.in.ImportCertARNs
	}
	return nil
```

Reply:
🤔 Those changes would give me nil pointer errors if those fields are nil.

Reply:
Wouldn't the above also result into a segfault if `e.in.ImportVPCConfig` is `nil`? I might be misunderstanding something!

What if we do:
```
if e.in.ImportVPCConfig != nil && len(e.in.ImportVPCConfig.PublicSubnetIDs) == 0
```